### PR TITLE
Migros crawler

### PR DESCRIPTION
**Pull request by kerembaskaya**

### CODE REVIEW

File: common/utils/mathematical_conversation_utils.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def convert_turkish_number_to_us(price: str) -> float:
```
Comment (kerembaskaya) [inline common/utils/mathematical_conversation_utils.py:1]: price_to_us

Comment (osmanbaskaya) [reply common/utils/mathematical_conversation_utils.py:1]: Bump

File: inflation-resources/tests/dataset/test_crawl.py [hunk 2]
```diff
@@ -5,31 +5,50 @@
...
-]
+from inflation.dataset.crawl import A101Crawler
+from inflation.dataset.crawl import MigrosCrawler
+
+CRAWLERS = {"a101": A101Crawler(), "migros": MigrosCrawler()}
+
+INFLATION_RESOURCES_PATH = Path(__file__).parents[2]
+
+TEST_PAGES_PATH = {
+    "a101": "tests/data/pages/ovadan-pirinc-baldo-1000-g-a101.html",
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_crawl.py:19]: Very nice

File: inflation-resources/tests/dataset/test_crawl.py [hunk 3]
```diff
@@ -5,31 +5,50 @@
...
+    "migros": "tests/data/pages/pinar-organik-sut-1L-migros.html",
+}
+
+RECORDS = {
+    "a101": [
+        ItemRecord("123", "pirinc", "ovadan", "http://www.a101.com", "a101")
+    ],
+    "a101-mig": [
+        ItemRecord("505", "sut", "pinar", "http://www.a101.com", "a101")
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_crawl.py:28]: Source is wrong. migros

File: inflation-resources/tests/dataset/test_crawl.py [hunk 4]
```diff
@@ -38,20 +57,34 @@ def inflation_data_reader():
 
 
 @pytest.fixture(scope="module")
-def html_test_file():
+def html_test_file_for_a101():
+    import codecs
+
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_crawl.py:62]: Bu importu yukarı al artik

File: inflation-resources/tests/dataset/test_crawl.py [hunk 5]
```diff
@@ -88,32 +125,69 @@ def test_crawl_output(mock_requests_get, crawler, html_test_file):
         assert test_data["item_code"] == "123"
         assert test_data["item_name"] == "pirinc"
         assert test_data["product_name"] == "ovadan"
-        assert test_data["source"] == "A101"
-        assert len(test_data["text"]) == 282375
+        assert test_data["source"] == "a101"
+        assert (
+            len(test_data["text"]) == 31721
+        )  # TODO: (KEREM) validate if this is correct.
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_crawl.py:131]: Todo silinsin mi?

File: inflation-resources/tests/dataset/test_crawl.py [hunk 6]
```diff
@@ -88,32 +125,69 @@ def test_crawl_output(mock_requests_get, crawler, html_test_file):
...
     """
+
+    mock_requests_get.return_value = mock.Mock(
+        status_code=200,
+        text=html_test_file_for_migros,  # TODO: I must change mock part since this use selenium
+    )
     with tempfile.TemporaryDirectory() as tmpdirname:
-        test_data_fp = crawler.crawl(RECORDS_RND, tmpdirname, "test")
+        test_data_fp = crawler_manager.start_crawling(
+            RECORDS["a101-mig"], tmpdirname
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_crawl.py:149]: a101-mig ne demek

File: inflation-resources/tests/dataset/test_crawl.py [hunk 7]
```diff
@@ -88,32 +125,69 @@ def test_crawl_output(mock_requests_get, crawler, html_test_file):
...
+        assert test_data["item_code"] == "505"
+        assert test_data["item_name"] == "sut"
+        assert test_data["product_name"] == "pinar"
+        assert test_data["source"] == "a101"
+        assert (
+            len(test_data["text"]) == 6561
+        )  # TODO(KEREM) validate if this is correct.
+
+
+# TODO: I'll add this part after re-writing reader parser
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_crawl.py:167]: Remove all this?

File: inflation/dataset/crawl.py [hunk 8]
```diff
@@ -145,9 +241,34 @@ def run():
...
-    crawler = Crawler()
-    records = crawler.parse_excel_to_link_dataset(file_path=args.excel_path)
-    crawler.crawl(records, path=args.path)
+    crawlers = {"a101": A101Crawler(), "migros": MigrosCrawler()}
+    cm = CrawlerManager(crawlers)
+    records = cm.parse_excel_to_link_dataset(file_path=args.excel_path)
+    cm.start_crawling(records, path=args.path)
+
+    # import argparse
+    #
```
Comment (osmanbaskaya) [inline inflation/dataset/crawl.py:250]: Don't leave any commented code

File: inflation/dataset/reader.py [hunk 9]
```diff
@@ -228,10 +235,79 @@ def read(self, filename, fields=None) -> InflationDataset:
...
+            print(subset_page_json["offers"]["availability"])
+            raise TypeError
+
+
 def run():
-    reader = InflationJSONA101DatasetReader()
-    dataset = reader.read("a101.med.json.gz")
-    print(dataset)
+    # reader = InflationJSONA101DatasetReader()
+    # dataset = reader.read("a101.med.json.gz")
```
Comment (osmanbaskaya) [inline inflation/dataset/reader.py:307]: Cleanup

File: inflation-resources/tests/dataset/test_crawl.py (hunk 3 above)
Comment (osmanbaskaya) [reply inflation-resources/tests/dataset/test_crawl.py:28]: @pytest.mark.xfail

File: inflation/dataset/crawl.py [hunk 10]
```diff
@@ -57,7 +176,7 @@ def parse_excel_to_link_dataset(self, file_path):
         Returns: List of TurkstatItemRecord
 
         """
-        file_path = format_spreadsheet_path(file_path)
+        file_path = Crawler.format_spreadsheet_path(file_path)
```
Comment (osmanbaskaya) [inline inflation/dataset/crawl.py:179]: move this function to CrawlerManager.

File: inflation/dataset/reader.py [hunk 11]
```diff
@@ -228,10 +235,79 @@ def read(self, filename, fields=None) -> InflationDataset:
...
+    def read_migros_page(self):  # url, driver_path, driver (quit demek icin)
+        driver = webdriver.Chrome(
+            executable_path="/Users/kerem/playground/kerem-side-projects-monorepo/common/chromedriver"
+        )
+        url = "https://www.migros.com.tr/pinar-organik-sut-1-l-p-a822f9"
+        driver.get(url)
+        try:
+            WebDriverWait(driver, 5).until(
+                EC.presence_of_element_located((By.CLASS_NAME, "price"))
+            )
```
Comment (osmanbaskaya) [inline inflation/dataset/reader.py:253]: sil bunu sil

**Pull request merged**